### ISSUE TITLE
make package install PROFILE=<your.bundle> error

### ISSUE THREAD

**Issue by hoangpx** [kabiroberai/theos-jailed#4]
```
Hoangs-MacBook-Pro:doodle hoangpx$ make package install PROFILE=com.sideload.DoodleJump
> Making all for tweak doodle…
make[2]: Nothing to be done for `internal-library-compile'.
> Making stage for tweak doodle…
==> Unpacking doodle.ipa…
==> Copying dependencies…
==> Injecting dependencies…
==> Signing DoodleJump.app…
==> Repacking DoodleJump.app…
==> Unpacking doodle-8+debug.ipa…
==> Waiting for iOS device to be connected…
==> Using J81AP 'iPad' (9515bd335d64804b9ac300497479eda53bc426ab).…
==> Installing: 70%
==> Error: Error 0xe8008015: A valid provisioning profile for this executable was not found. AMDeviceSecureInstallApplication(0, device, url, options, install_callback, 0)
```
Here is the log when running `make clean package messages=yes`
[(https://ghostbin.com/paste/dpcd9)]

I tried to install the package named **doodle-8+debug.ipa** I got in packages folder via xcode but there's an another error **App installation failed. A signed resource has been added, modified, or deleted.**

Finally, I use `make package PROFILE=com.sideload.DoodleJump CODESIGN_IPA=0` to to disable codesigning completely then user Cydia Impactor to install, got another error **ipa.cpp:263
ldid/ldid.cpp(1910): _assert(plist != __null)**

I also install this package via xcode then I got error **App installation failed. An unknown error has occurred.**

**Comment by kabiroberai** [comment]
Are you sure the profile is installed on your device? If it is and still doesn't work, try resigning with [iOS App Signer](https://dantheman827.github.io/ios-app-signer/) and then installing that via Cydia Impactor.

**Comment by hoangpx** [comment]
I'm sure the profile is installed on my devices. Yes I can resign it with Signer then install it likes normal. But does this means st wrong with your module or ios-deploy ?

**Comment by kabiroberai** [comment]
It's probably the module. I'll look into the issue in a while

**Comment by kabiroberai** [comment]
I think I might have fixed this with 0d947a70a7137dcc1ea3dceac746077c6dbd85b3. Try running `make update-jailed`, and let me know if it works after that.

**Comment by kabiroberai** [comment]
How many developer certificates do you have? If you have multiple, it may be codesigning with the incorrect one. Please run `security find-certificate -ac "iPhone Developer" | grep alis | cut -f4 -d\" | cut -f1 -d\(` and paste the output.

**Comment by kabiroberai** [comment]
Hey, apparently the `_assert(plist != __null)` issue [has been fixed](https://www.reddit.com/r/sideloaded/comments/5jrxq6/question_can_someone_help_me_with_this_error/dbiomgm/?) in Cydia Impactor 0.9.36. Could you please update impactor and see if it works that way? Also, what was the result of my previous suggestion?

**Comment by hoangpx** [comment]
the output 
iPhone Developer: a*** 
iPhone Developer: b***
iPhone Developer: c***
iPhone Developer: d***

**Comment by kabiroberai** [comment]
Which of the 4 teams did you use to create the `com.sideload.DoodleJump` profile? Can you set `DEV_CERT_NAME = iPhone Developer: Your Team` at the top of `Makefile`?

**Comment by hoangpx** [comment]
nothing different 
`make package PROFILE=com.sizeload.zingmp3-lite
> Making all for tweak mp3…
make[2]: Nothing to be done for `internal-library-compile'.
> Making stage for tweak mp3…
==> Unpacking mp3.ipa…
==> Copying dependencies…
==> Injecting dependencies…
==> Signing MediaiOS.app…
==> Error: Failed to find profile for 'com.sizeload.zingmp3-lite'
make: *** [internal-package] Error 1`

**Comment by hoangpx** [comment]
nothing different 

> make package PROFILE=com.sizeload.zingmp3-lite
> > Making all for tweak mp3…
> make[2]: Nothing to be done for `internal-library-compile'.
> > Making stage for tweak mp3…
> ==> Unpacking mp3.ipa…
> ==> Copying dependencies…
> ==> Injecting dependencies…
> ==> Signing MediaiOS.app…
> ==> Error: Failed to find profile for 'com.sizeload.zingmp3-lite'
> make: *** [internal-package] Error 1

**Comment by hoangpx** [comment]
<img width="900" alt="screen shot 2016-12-24 at 9 38 09 pm" src="https://cloud.githubusercontent.com/assets/1779511/21469737/55a0b3e6-ca21-11e6-91f1-bcb8d99961c7.png">

**Comment by hoangpx** [comment]
<img width="840" alt="screen shot 2016-12-24 at 9 39 16 pm" src="https://cloud.githubusercontent.com/assets/1779511/21469738/757af0be-ca21-11e6-83c0-0c75c016a46a.png">

**Comment by hoangpx** [comment]
here is ipa file I'm trying to tweak, I decrypted it by Clutch. 
[mp3.ipa.zip](https://github.com/kabiroberai/theos-jailed/files/672060/mp3.ipa.zip)

**Comment by hoangpx** [comment]
I also trying to use CODESIGN_IPA=0, then resign by  iOS App Signer and use latest (0.9.37) Cydia Impactor to install but always get error `ipa.cpp:348
ldid/ldid.cpp(1938): _assert(plist != __null)` .

**Comment by kabiroberai** [comment]
Seems you spelled `sideload` as `sizeload`. Could that be it?

**Comment by kabiroberai** [comment]
Can you `make troubleshoot PROFILE=com.sideload.zingmp3-lite` again?

**Comment by hoangpx** [comment]
I corrected sideload. But still 
https://ghostbin.com/paste/9u8wq

**Comment by kabiroberai** [comment]
Sorry, `make troubleshoot-jailed`

**Comment by hoangpx** [comment]
https://ghostbin.com/paste/c5c6e

**Comment by kabiroberai** [comment]
Also try `make clean do all messages=yes PROFILE=com.sideload.zingmp3-lite DEV_CERT_NAME="iPhone Developer: Thu Trang"`, with your device connected

**Comment by hoangpx** [comment]
<img width="843" alt="screen shot 2016-12-24 at 10 08 57 pm" src="https://cloud.githubusercontent.com/assets/1779511/21469804/97998c7e-ca25-11e6-9bbd-7e50d514f3ed.png">

**Comment by kabiroberai** [comment]
Try enabling **App Groups** in Capabilities, and then running the same command

**Comment by kabiroberai** [comment]
It seems that the entitlements file specifies a `production` aps-environment, with `get-task-allow` disabled. I think that could be the issue. Could you try downloading a decrypted ipa online and seeing if that works?

**Comment by hoangpx** [comment]
<img width="1392" alt="screen shot 2016-12-24 at 10 18 57 pm" src="https://cloud.githubusercontent.com/assets/1779511/21469834/17aefa1a-ca27-11e6-87ee-1dea5fbbad7e.png">
Checked, still the same, do I need to add it to some group ?

**Comment by hoangpx** [comment]
oh. It works on the app downloaded from the Internet. So What should I do with my ownapp ? Could you give me some directions ?

**Comment by kabiroberai** [comment]
I'm not sure, since I don't have a jailbroken device atm. Maybe try using [dumpdecrypted](https://github.com/stefanesser/dumpdecrypted) instead of Clutch?

**Comment by hoangpx** [comment]
same. I tried to insert Entitlements with get-task-allow = true but no luck. I will keep updating here.

**Comment by kabiroberai** [comment]
Commit 2ac06652f078e2c7f1d6f6147ee0c7bd7550a1b7 should have fixed this issue. You need to do one extra step though, set `BUNDLE_ID = com.sideload.zingmp3-lite` in `Makefile`. Once done, it should (hopefully) work.

**Comment by kabiroberai** [comment]
GitHub automatically closed the issue due to the commit message, but I'll reopen it if the problem isn't resolved.

**Comment by hoangpx** [comment]
same. I think there's some thing wrong in injecting dylib into ipa file ...
`Hoangs-MacBook-Pro:mediaios hoangpx$ make clean do all messages=yes PROFILE=com.sideload.zingmp3-lite DEV_CERT_NAME="iPhone Developer: Thu Trang"
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 6 "Cleaning"); set -o pipefail; (rm -rf "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj")
==> Cleaning…
set -o pipefail; (rm "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/build_session")
set -o pipefail; (touch "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/build_session")
set -o pipefail; (rm -rf "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_")
> Making all for tweak MediaiOS…
set -o pipefail; (/Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile --no-print-directory --no-keep-going \
		internal-library-compile \
		_THEOS_CURRENT_TYPE=tweak THEOS_CURRENT_INSTANCE=MediaiOS _THEOS_CURRENT_OPERATION=compile \
		THEOS_BUILD_DIR="." _THEOS_MAKE_PARALLEL=yes)
set -o pipefail; (mkdir -p /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/)
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 1 "Preprocessing Tweak.xm"); set -o pipefail; (/usr/local/bin/theos/bin/logos.pl -c warnings=error    Tweak.xm > /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/Tweak.xm.mm)
==> Preprocessing Tweak.xm…
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 2 "Compiling Tweak.xm (armv7)"); set -o pipefail; (xcrun -sdk iphoneos clang++ -x objective-c++ -c -I"/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios" -I/usr/local/bin/theos/include -include /usr/local/bin/theos/Prefix.pch -DTARGET_IPHONE=1 -O0 -Wall -Werror  -isysroot "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.2.sdk" -D__IPHONE_OS_VERSION_MIN_REQUIRED=__IPHONE_6_0 -miphoneos-version-min=6.0  -fmodules -fcxx-modules -fmodule-name=MediaiOS -fbuild-session-file=/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/build_session -fmodules-prune-after=345600 -fmodules-prune-interval=86400 -fmodules-validate-once-per-build-session   -I/usr/local/bin/theos/vendor/include -I/usr/local/bin/theos/include/_fallback -DDEBUG -ggdb -O0  -DTHEOS_INSTANCE_NAME="\"MediaiOS\""  -arch armv7              /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/Tweak.xm.mm -o /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/Tweak.xm.3c36a6b3.o)
==> Compiling Tweak.xm (armv7)…
set -o pipefail; (rm /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/Tweak.xm.mm)
set -o pipefail; (mkdir -p /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7)
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 3 "Linking tweak MediaiOS (armv7)"); set -o pipefail; (xcrun -sdk iphoneos clang++ -L/usr/local/bin/theos/lib -F/usr/local/bin/theos/mod/jailed/lib -L/usr/local/bin/theos/vendor/lib -lobjc -framework Foundation -framework CoreFoundation -ObjC++ -fobjc-exceptions -fobjc-call-cxx-cdtors              -stdlib=libc++ -F/usr/local/bin/theos/vendor/lib -framework CydiaSubstrate -dynamiclib -install_name "/Library/MobileSubstrate/DynamicLibraries/MediaiOS.dylib"  -isysroot "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.2.sdk" -D__IPHONE_OS_VERSION_MIN_REQUIRED=__IPHONE_6_0 -miphoneos-version-min=6.0 -Wl,-segalign,4000 -multiply_defined suppress -arch armv7  -ggdb -O0 -o "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/MediaiOS.dylib" /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/Tweak.xm.3c36a6b3.o | (grep -v 'usr/lib/dylib1.o, missing required architecture' || true))
==> Linking tweak MediaiOS (armv7)…
set -o pipefail; (mkdir -p /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/)
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 1 "Preprocessing Tweak.xm"); set -o pipefail; (/usr/local/bin/theos/bin/logos.pl -c warnings=error    Tweak.xm > /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/Tweak.xm.mm)
==> Preprocessing Tweak.xm…
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 2 "Compiling Tweak.xm (arm64)"); set -o pipefail; (xcrun -sdk iphoneos clang++ -x objective-c++ -c -I"/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios" -I/usr/local/bin/theos/include -include /usr/local/bin/theos/Prefix.pch -DTARGET_IPHONE=1 -O0 -Wall -Werror  -isysroot "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.2.sdk" -D__IPHONE_OS_VERSION_MIN_REQUIRED=__IPHONE_6_0 -miphoneos-version-min=6.0  -fmodules -fcxx-modules -fmodule-name=MediaiOS -fbuild-session-file=/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/build_session -fmodules-prune-after=345600 -fmodules-prune-interval=86400 -fmodules-validate-once-per-build-session   -I/usr/local/bin/theos/vendor/include -I/usr/local/bin/theos/include/_fallback -DDEBUG -ggdb -O0  -DTHEOS_INSTANCE_NAME="\"MediaiOS\""  -arch arm64              /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/Tweak.xm.mm -o /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/Tweak.xm.041cec7c.o)
==> Compiling Tweak.xm (arm64)…
set -o pipefail; (rm /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/Tweak.xm.mm)
set -o pipefail; (mkdir -p /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64)
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 3 "Linking tweak MediaiOS (arm64)"); set -o pipefail; (xcrun -sdk iphoneos clang++ -L/usr/local/bin/theos/lib -F/usr/local/bin/theos/mod/jailed/lib -L/usr/local/bin/theos/vendor/lib -lobjc -framework Foundation -framework CoreFoundation -ObjC++ -fobjc-exceptions -fobjc-call-cxx-cdtors              -stdlib=libc++ -F/usr/local/bin/theos/vendor/lib -framework CydiaSubstrate -dynamiclib -install_name "/Library/MobileSubstrate/DynamicLibraries/MediaiOS.dylib"  -isysroot "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.2.sdk" -D__IPHONE_OS_VERSION_MIN_REQUIRED=__IPHONE_6_0 -miphoneos-version-min=6.0  -multiply_defined suppress -arch arm64  -ggdb -O0 -o "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/MediaiOS.dylib" /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/Tweak.xm.041cec7c.o | (grep -v 'usr/lib/dylib1.o, missing required architecture' || true))
==> Linking tweak MediaiOS (arm64)…
(printf "\e[0;3%im==> \e[1;39m%s…\e[m\n" 4 "Merging tweak MediaiOS"); set -o pipefail; (xcrun lipo -arch armv7 /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/armv7/MediaiOS.dylib -arch arm64 /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/arm64/MediaiOS.dylib -create -output "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/obj/debug/MediaiOS.dylib")
==> Merging tweak MediaiOS…
set -o pipefail; (rm -rf "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_")
set -o pipefail; (/usr/local/bin/theos/bin/fakeroot.sh -p "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/fakeroot" -c)
set -o pipefail; (mkdir -p "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_")
set -o pipefail; ([ -d /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/layout ] && rsync -a "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/layout/" "/Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_" --exclude "DEBIAN" --exclude "_MTN" --exclude ".git" --exclude ".svn" --exclude ".DS_Store" --exclude "._*" || true)
> Making stage for tweak MediaiOS…
set -o pipefail; ("/usr/local/bin/theos/mod/jailed/bin/ipa.sh")
==> Unpacking mp3.ipa…
==> Copying dependencies…
==> Injecting dependencies…
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: for: /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_/Payload/MediaiOS.app/MediaiOS (for architecture arm64) option "-add_rpath @executable_path/Frameworks" would duplicate path, file already has LC_RPATH for: @executable_path/Frameworks
LC_CODE_SIGNATURE load command found. Remove it? [y/n] y
Added LC_LOAD_DYLIB to /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_/Payload/MediaiOS.app/MediaiOS
Added LC_LOAD_DYLIB to /Users/hoangpx/OngoingProjects/Zingmp3/mp3/mediaios/.theos/_/Payload/MediaiOS.app/MediaiOS
==> Signing MediaiOS.app…
==> Error: Failed to find profile for 'com.sideload.zingmp3-lite'
make: *** [internal-package] Error 1`

**Comment by kabiroberai** [comment]
That's an error with finding the profile, not codesigning or injecting. What's the result of `grep -ahr "iOS Team Provisioning Profile: " ~/Library/MobileDevice/Provisioning\ Profiles`?

**Comment by hoangpx** [comment]
It's work ! Thanks a lot for supporting me. I still have to include `DEV_CERT_NAME="iPhone Developer: yourname"`

**Comment by kabiroberai** [comment]
Welcome! I'm currently trying to make it so that dev_cert_name is automatically extracted from the profile, so expect to see that in the next commit.

**Comment by kabiroberai** [comment]
Done. Run `make update-jailed` and you should be able to remove `DEV_CERT_NAME`.

**Comment by hoangpx** [comment]
Tested. Question : After build and deploying on my first device, I used Cydia Impactor to install the ipa file from packages folder to another device but always getting error ipa.cpp:348
<img width="532" alt="screen shot 2016-12-26 at 10 29 12 pm" src="https://cloud.githubusercontent.com/assets/1779511/21491540/c1b7438e-cbba-11e6-9070-2c486fdcc948.png">
Couldn't find any clue on google

**Comment by hoangpx** [comment]
I tried resign that ipa with Singer but still get that error

**Comment by hoangpx** [comment]
I even get ipa file from first jailbroken device, then use Impactor/Signer to install it on second nonjailbroken device, but no luck. This method I use for running paid app on nonjailbroken device and it always works, so I think something changed in ipa file or st related to distribution stuffs

**Comment by kabiroberai** [comment]
For some reason (no clue why), Impactor doesn't like CydiaSubstrate too much. You can make it work by setting `TWEAK_NAME_USE_SUBSTRATE=0`, replacing `TWEAK_NAME` with your tweak's name.

**Comment by hoangpx** [comment]
It works. Awesome.

**Comment by kabiroberai** [comment]
Hey, I'm reopening this until I figure out why CydiaSubstrate doesn't mix well with Impactor. I've private messaged Saurik on Reddit, so hopefully he should be able to help resolve this issue soon.

**Comment by kuteteen** [comment]
@hoangpx có email ko cho mình nhờ chút với

**Comment by kuteteen** [comment]
==> Error: Could not find profile '*'
make: *** [internal-package] Error 1

**Comment by kuteteen** [comment]
use UIAlertView in tweak.xm not working?? help me fix please

**Comment by kabiroberai** [comment]
@hoangpx I've solved the issue with 67c6d3b0a553c217ee402bb015860a7b93014eab. Run `make update-jailed`, and tweaks that use CydiaSubstrate should work correctly with Impactor.

**Comment by hoangpx** [comment]
@kuteteen add my skype: hoangpx
